### DISCUSSION TITLE
Overriding messages (scope question)

### DISCUSSION THREAD

**Discussion by marr** [kazupon/vue-i18n#1530] [Q&A]
I am using vue 2 and version 8 of vue-i18n, and having trouble overriding messages. Is it possible to:

1. have a "global" set of messages in for example _locale/en.json_ 
2. use those messages as defaults
3. override messages in a component, say _App.vue_ and have any components used there use that override.

I have a repo that has this general reproduction case:
https://github.com/marr/vue2-i18n

You can see that the messages defined in _App.vue_ **do** apply in direct usage of `$t` in the file, but not with components on that page that use those same messages.

_App.vue_:
```
i18n: { messages: { en: { hello: 'hi' } } }
...
$t('hello') === 'hi'
<Hello /> === 'default'
```
_Hello.vue_
```
$t('hello')
```

**Comment by johnantoni** [comment]
I'd like this to, especially for component-based translations. So far haven't found a solution.

Maybe having a handler that sits in-between $t might work for now till a official solution appears
